scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -1613,3 +1613,246 @@ cdef class DiscreteAliasUrn(Method):
+    function proportional to the density is used). The equality holds since
+    the area of `A` is equal to 0.5 * area(pdf) (Theorem 7.1 in [4]_).
+
+    In the general case, the acceptance region is defined as
+    ``A = {(u, v) : 0 < v <= (pdf(u/v**r + center))**(1/(r+1))}`` and the
+    bounding rectangle is given by
+
+    - ``v_max = sup_x pdf(x)**(1/(r+1))``
+    - ``u_min = inf_x (x - center) (pdf(x))**(r/(r+1))``
+    - ``u_max = sup_x (x - center) (pdf(x))**(r/(r+1))``
+
+    References
+    ----------
+    .. [1] UNU.RAN reference manual, Section 5.3.11,
+           "NROU - Naive Ratio-Of-Uniforms method",
+           http://statmath.wu.ac.at/software/unuran/doc/unuran.html#NROU
+    .. [2] W. Hörmann, J. Leydold and G. Derflinger (2004). Automatic
+           Nonuniform Random Variate Generation, Springer, Berlin.
+    .. [3] A.J. Kinderman and J.F. Monahan, "Computer Generation of Random
+           Variables Using the Ratio of Uniform Deviates",
+           ACM Transactions on Mathematical Software, 3(3), p. 257--260, 1977.
+    .. [4] L. Devroye, "Non-Uniform Random Variate Generation",
+           Springer-Verlag, 1986.
+    .. [5] J. C. Wakefield, A. E. Gelfand, and A. F. M. Smith. "Efficient
+           generation of random variates via the ratio-of-uniforms method."
+           Statistics and Computing 1.2, p. 129--133, 1991.
+
+    Examples
+    --------
+    >>> from scipy.stats import NaiveRatioUniforms
+
+    Suppose we want to generate normally distributed random variables.
+    It is easy to compute the bounding rectangle explicitly in that case.
+    Note that we drop the normalization factor of the density. Since the mode
+    of the distribution is located at zero, there is no need to change the
+    default value of the parameter `center`.
+
+    >>> urng = np.random.default_rng()
+    >>> class MyDist:
+    ...     def pdf(self, x):
+    ...         return np.exp(-0.5*x*x)
+    >>> dist = MyDist()
+    >>> u_bound = np.sqrt(dist.pdf(np.sqrt(2))) * np.sqrt(2)
+    >>> v_max = np.sqrt(dist.pdf(0))
+    >>> rng = NaiveRatioUniforms(dist, u_min=-u_bound, u_max=u_bound,
+    ...                          v_max=v_max, random_state=urng)
+
+    Once we have created the generator, we can draw samples of the normal
+    distribution:
+
+    >>> import matplotlib.pyplot as plt
+    >>> rvs = rng.rvs(1000)
+    >>> x = np.linspace(rvs.min()-0.1, rvs.max()+0.1, 100)
+    >>> fx = dist.pdf(x) / np.sqrt(2*np.pi)
+    >>> plt.plot(x, fx, 'r-', lw=2, label='normal pdf')
+    >>> plt.hist(rvs, bins=20, density=True, alpha=0.8, label='random variates')
+    >>> plt.xlabel('x')
+    >>> plt.legend()
+    >>> plt.show()
+
+    """
+
+    def __cinit__(self,
+                  dist,
+                  center=0.0,
+                  *,
+                  domain=None,
+                  r=1.0,
+                  u_min,
+                  u_max,
+                  v_max,
```

Comment:
Maybe change this to:

```suggestion
                  u_min=None,
                  u_max=None,
                  v_max=None,
```

I checked locally and UNU.RAN automatically calculates these using numerical search if not given by the user. So, it would be nice to keep these parameters optional.